lib/taurus/qt/qtgui/base/taurusbase.py

```diff
@@ -1342,6 +1361,8 @@ def __init__(self, name='', parent=None, designMode=False, **kwargs):
         self.call__init__(TaurusBaseComponent, name,
                           parent=parent, designMode=designMode)
         self._setText = self._findSetTextMethod()
+        self._formaters = scheme_formatters
+        self._formaters['{:2.3e}'] = '{:2.3e}'
```

Comment:
Similar to what I was mentioning before: I would not keep a list of loaded formatters referenced by the TaurusBaseWidget. In other words, instead of keeping `self._formaters` I would just call the getter whenever is needed (e.g. in `showFormatterDlg()`)

Reply:
After discussing with @cmft , we agreed that keeping a list may still be needed in order to offer persistence in the choice of formatter hwen the user sets a custom formatter